blocks/bricks/sequence_generators.py

```diff
@@ -290,38 +290,81 @@ def readout(self, **kwargs):
     readout_dim : int
         The dimension of the readout.
     emitter : an instance of :class:`AbstractEmitter`
         The emitter component.
     feedback_brick : an instance of :class:`AbstractFeedback`
         The feedback component.
+    merger : :class:`.Brick`, optional
+        A brick that takes the sources given in `source_names` as an input
+        and combines them into a single output. If given, `merge_prototype`
+        cannot be given.
+    merge_prototype : :class:`.FeedForward`, optional
+        If `merger` isn't given, the transformation given by
+        `merge_prototype` is applied to each input before being summed. By
+        default a :class:`.Linear` transformation without biases is used.
+        If given, `merger` cannot be given.
+    post_merge : :class:`.Feedforward`, optional
+        If given, this transformation is applied to the merged inputs to
+        the readout model.
+    post_merge_input_dim : int, optional
+        The input dimension of `post_merge` i.e. the output dimension of
+        `merger` (or `merge_prototype`).
 
     """
     @lazy
-    def __init__(self, readout_dim=None, emitter=None, feedback_brick=None,
-                 **kwargs):
+    def __init__(self, source_names, readout_dim, emitter=None,
+                 feedback_brick=None, merger=None, merge_prototype=None,
+                 post_merge=None, post_merge_input_dim=None, **kwargs):
         super(Readout, self).__init__(**kwargs)
+        self.source_names = source_names
         self.readout_dim = readout_dim
 
         if not emitter:
             emitter = TrivialEmitter(readout_dim)
         if not feedback_brick:
             feedback_brick = TrivialFeedback(readout_dim)
+        if not merger:
+            merger = Merge(input_names=source_names, prototype=merge_prototype)
         self.emitter = emitter
         self.feedback_brick = feedback_brick
+        self.merger = merger
+        self.post_merge = post_merge
+        self.post_merge_input_dim = post_merge_input_dim
 
-        self.children = [self.emitter, self.feedback_brick]
+        self.children = [self.emitter, self.feedback_brick, self.merger,
+                         self.post_merge]
 
     def _push_allocation_config(self):
         self.emitter.readout_dim = self.get_dim('readouts')
         self.feedback_brick.output_dim = self.get_dim('outputs')
+        self.merger.input_names = self.source_names
+        self.merger.input_dims = {source_name: self.source_dims[source_name]
+                                  for source_name in self.source_names}
+        if self.post_merge:
+            self.post_merge.output_dim = self.readout_dim
+            self.post_merge.input_dim = self.post_merge_input_dim
+            self.merger.output_dim = self.post_merge_input_dim
+        else:
+            self.merger.output_dim = self.readout_dim
+
+    @application
+    def readout(self, **kwargs):
+        if set(kwargs.keys()) != set(self.merger.input_names):
```

Comment:
This is a nice thing to do in general, but unfortunately it will receive all possible sources as `kwargs`. In early days of Blocks it was not clear to me who should filter necessary input, the calling brick of the callee. Now I think that the calling brick.

Could you please use `dict_dubset` where `readout` is used, that is in `cost` and `generate` methods?